[DS-5477] Adapt the app to be used with Valkey

shirazJafri-a9s opened the pull request.

CedrikHeusser commented on main.go:
```diff
@@ -44,78 +52,109 @@ func initTemplates() {
 
-func createCredentials() (RedisCredentials, error) {
+func createCredentials() (ValkeyCredentials, error) {
 	// Kubernetes
```
Although unchanged, this comment is misleading, or? I would use something like `local` or smth similar. Or do I misunderstand it entirely? :D

shirazJafri-a9s replied:
This was an oversight on my part, nice catch!

Commit:
Revamp application to be used with Valkey
- Bump Go to 1.23 and use Valkey client libraries.
- Bump dependencies.
- Add TLS functionality but keep certificate optional.
- Remove Redis client code and any mention of Redis in CSS files.
- Update `README.md` for Valkey.
- Update Dockerfile to use Valkey.
- Remove `kubernetes-deployment.yml`
